src/replicache-options.ts

```diff
@@ -37,14 +37,20 @@ export interface ReplicacheOptions<MD extends MutatorDefs> {
   pullURL?: string;
 
   /**
-   * The name of the Replicache database.
+   * A unique identifier for the user authenticated by
+   * [[ReplicacheOptions.auth]]. Must be non-empty.
    *
-   * It is important to use user specific names so that if there are multiple
-   * tabs open for different distinct users the data is kept seperate.
+   * This is used to keep different user's state separate.
    *
-   * You can use multiple Replicache instances as long as the names are unique.
+   * For efficiency, a new Replicache instance will initialize its state from
+   * the persisted state of an existing Replicache instance with the same
+   * `userID`, domain and browser profile.
+   *
+   * Mutations from one Replicache instance may be pushed using the
+   * [[ReplicacheOptions.auth]] of another Replicache instance with the same
+   * `userID`, domain and browser profile.
    */
-  name: string;
```

Comment:
I deleted rather than deprecating ReplicacheOptions.name because I want to make userID required, in which case it doesnt make sense to still have a name option.

I deprecated Replicache.name, and assigned it the value of `userID`.

What do you think @arv ?

Reply:
I prefer to just make it a breaking change cleanly. 9.0 is already a breaking version and it's important people understand the semantics of this field.